clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -565,19 +555,16 @@ public void testInitTransactionTimeout() {
         Node node = cluster.nodes().get(0);
 
         Metadata metadata = new Metadata(0, Long.MAX_VALUE, true);
-        metadata.update(cluster, Collections.<String>emptySet(), time.milliseconds());
+        metadata.update(cluster, Collections.emptySet(), time.milliseconds());
 
         MockClient client = new MockClient(time, metadata);
         client.setNode(node);
 
-        Producer<String, String> producer = new KafkaProducer<>(
+        try (Producer<String, String> producer = new KafkaProducer<>(
                 new ProducerConfig(ProducerConfig.addSerializerToConfig(props, new StringSerializer(), new StringSerializer())),
-                new StringSerializer(), new StringSerializer(), metadata, client);
-        try {
+                new StringSerializer(), new StringSerializer(), metadata, client)) {
             producer.initTransactions();
             fail("initTransactions() should have raised TimeoutException");
-        } finally {
-            producer.close(0, TimeUnit.MILLISECONDS);
```

Comment:
This seems to change the behaviour...

Reply:
Indeed it changes the behavior of how we close the producer as it won't be a force close after the refactor, however I don't think it changes the behavior of the test itself. Since we're not setting any response on the mock client, the initTransaction should timeout regardless of how do we close the connection at the end.
@huxihx as you're the author of the original code, would you please help us whether this change is correct or are there any details I'm missing?